core/src/main/java/com/graphhopper/storage/CHGraphImpl.java

```diff
@@ -483,11 +466,11 @@ public void debugPrint() {
     }
 
     private int getNodeA(long edgePointer) {
-        return shortcuts.getInt(edgePointer + E_NODEA);
+        return shortcuts.getInt(edgePointer + E_NODEA) >>> 1;
```

Comment:
no need to convert to long and do the mask with 0xFFFFFFFFL as we do for the weight value, because we only use 31 bits that fit into an int (after we do the unsigned shift)